src/sql/statements.cpp

```diff
@@ -208,6 +215,34 @@ namespace hsql {
       }
       delete order;
     }
+
+    if (setStatement != nullptr) {
+      for (SelectStatement* stmt : *setStatement) {
+        delete stmt;
+      }
+      delete setStatement;
+    }
+
+    if (setType != nullptr) {
+      for (SetDescription* desc : *setType) {
+        delete desc;
+      }
+      delete setType;
+    }
+  }
+
+  SelectStatement* MakeOrAppendUnionList(SelectStatement* stmt,
```

Comment:
It's not only a UnionList, is it? Maybe MakeOrAppendSetList?

Reply:
I have renamed to `MakeOrAppendSetList` as suggested